cachito/workers/pkg_managers/general.py

```diff
@@ -187,8 +187,17 @@ async def async_download_binary_file(
                     if not chunk:
                         break
                     f.write(chunk)
-    except Exception as e:
-        raise NetworkError(f"Could not download {url}: {e}")
+
+    except Exception as exception:
+        log.error(f"Unsuccessful download: {tarball_name}")
+        message = (
+            f"Could not download {tarball_name} from {url}. "
+            f"exception_name: {exception.__class__.__name__}, "
+            f"details: {exception}"
+        )
+
+        raise NetworkError(message) from None
+        # Using "from None" since we have the exception context in the logs
```

Comment:
If we keep the more verbose err. msg and suppress the context, could the comment go above the `raise` line? I completely missed it at first when looking at the diff and was wondering why the conversation was resolved :smile:

Reply:
don't we usually write comments on line before?

https://peps.python.org/pep-0008/#block-comments

Reply:
what adam said :D

Reply:
Sorry, I put the cart before the horse :facepalm: 

![image](https://user-images.githubusercontent.com/29442029/199727616-9d25635e-c28c-406f-8227-9a55f9cac324.png)